.gitlab-ci.yml

```diff
@@ -0,0 +1,65 @@
+# This file is a template, and might need editing before it works on your project.
+image: golang:1.11.4
+
+variables:
+  # Please edit to your GitLab project
+  REPO_NAME: github.com/mendersoftware/mender
+
+before_script:
+  - mkdir -p /go/src/$(dirname $REPO_NAME)/mender /go/src/_/builds
+  - cp -r $CI_PROJECT_DIR /go/src/$(dirname $REPO_NAME)
+  # - ln -s /go/src/$(dirname $REPO_NAME) /go/src/_/builds/mendersoftware
+  - cd $GOPATH/src/$REPO_NAME
+  - apt-get update && apt-get install -yyq liblzma-dev
+  - make get-tools
+
+stages:
+  - test
+  - build
+  - publish
+
+test:format:
+  stage: test
+  allow_failure: true
+  script:
+    - go fmt $(go list ./... | grep -v /vendor/)
+    - go vet $(go list ./... | grep -v /vendor/)
+    - go test -race $(go list ./... | grep -v /vendor/)
+
+test:checks:
+  stage: test
+  script:
+    - make extracheck
+    - make coverage
+    - mkdir -p tests/unit-coverage && find . -name 'coverage.txt' -exec cp --parents {} ./tests/unit-coverage \;
+    - tar -cvf $CI_PROJECT_DIR/unit-coverage.tar tests/unit-coverage
+  tags:
+    - mender-qa-slave
+  artifacts:
+    expire_in: 30min
+    paths:
+      - unit-coverage.tar
+
+compile:
+  stage: build
+  script:
+    - make build V=1
+    - cp mender $CI_PROJECT_DIR
+  artifacts:
+    paths:
+      - mender
+
+publish:tests:
+  stage: publish
+  dependencies:
+    - test:checks
+  script:
+    - tar -xvf unit-coverage.tar
+    # - bash <(curl -s https://codecov.io/bash) -s ./tests/unit-coverage;
```

Comment:
At some point we will do this from GitLab. Do you think we should create a task for it (for all repos) as a reminder?

Reply:
Do you mean that coverage analysis is part of Gitlab too?

Reply:
The approach is that what today is done by Travis (unit testing, code coverage, etc) will be openly done by Gitlab in each repo individual Pipeline, and what it is done by Jenkins (mender QA) will be closely done by Gitlab in mender QA Pipeline.

This can still be further discussed, but it seems natural to me to not maintain Travis build recipes after the migration.

Reply:
Apart from what @lluiscampos said, you can only get the overall percentage: https://docs.gitlab.com/ee/user/project/pipelines/settings.html#test-coverage-parsing in a badge https://docs.gitlab.com/ee/user/project/pipelines/settings.html#test-coverage-report-to show in the repo, but the analysis for the PR (as done by codecov.io) does not exist from Gitlab

@lluiscampos ticket created (QA-81) - but this should wait until we are convinced

Reply:
Yes, my point was whether we would still use codecov.io. And I guess we will, if Gitlab only offers one simple number.